### DISCUSSION TITLE
multi language files

### DISCUSSION THREAD

**Discussion by gdkrmr** [helix-editor/helix#9004] [Q&A]
I see that markdown files use correct syntax highlighting for code chunks by default, which I find great. E.g.

````
```{r}
x <- 1
```
````

gets highlighted correctly. How does this work with language servers? I have a language server for R that works in .R files but I had to add the R language server manually for markdown files.

```toml
[[language]]
name = "markdown"
file-types = ["Rmd","qmd", "md"]
language-servers = ["r", "julia-lsp", "markdown"]
```

is there a better way to configure this?

**Comment by the-mikedavis** [comment]
The LSP spec doesn't currently support running a server on a fragment of a document - all documents must be backed by files. Sometimes a language server will support handling multiple languages (for example an HTML language server might support JavaScript in `script` tags) but there isn't a way for us to dynamically dispatch parts of each document to different language servers.
